### PR TITLE
Delete Reviews

### LINKED ISSUE #16: User Story 18, User can delete a book review
```
As a Visitor,
When I visit a user's show page,
I see a link next to each review to delete the review.
When I delete a review I am returned to the user's show page
Then I should no longer see that review.
```

DO THESE STORIES FIRST:
- User Story 13, Add a Review for a Book
- User Story 16, User Show Page

### Checklist:

- [x] base code tests are written
- [x] base code is written to pass tests
- [ ] edge case ("sad path") tests written
- [ ] edge case code written
- [x] code is reviewed
- [x] branch is merged
- [x] all tests still pass

child of #9

child of #33

depends on #16

depends on #11

### PR DESCRIPTION
**Pull request by BrennanAyers**
This PR brings in the functionality to remove a Review from the database from our website. The option to delete is only present on the User who made the Review's show page.
Upon clicking the delete link, they will be given a warning and confirmation of their deletion. Upon confirming, they will be sent back to that User show page, where they will find the Review removed.
Since the Review table is a joins table between Books and Users, the Review will no longer show on the Book as well.

Resolves #16

### CODE REVIEW

File: app/controllers/reviews_controller.rb [hunk 1]
```diff
@@ -14,6 +14,14 @@ def create
     redirect_to books_path(book)
   end
 
+  def destroy
+    # This is very confusing, but our Review ID is passed into params as :book_id because Reviews are nested resources of Book
+    deleted_review = Review.find(params[:book_id])
+    deleted_review.destroy
```
Comment (BrennanAyers) [inline app/controllers/reviews_controller.rb:20]: As the comment says, this is a big counter-intuitive, but I'm not entirely sure how to avoid it considering our Reviews are nested within Books.

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -12,6 +12,7 @@
       <h5><%= review.title %></h5>
       <p>Rating: <%= review.rating %></p>
       <p>Review: <%= review.text %></p>
+      <%= link_to "Delete Review", book_review_path(review), method: :delete, data: {confirm: "Are you sure you want to delete the review?"} %>
```
Comment (BrennanAyers) [inline app/views/users/show.html.erb:15]: Will need to implement this link_to in other portions as well.
